src/software/world/team.cpp

```diff
@@ -39,7 +39,7 @@ void Team::updateRobots(const std::vector<Robot>& new_robots)
         }
 
         auto it = std::find_if(team_robots.begin(), team_robots.end(),
-                               [&](const Robot& r) { return r.id() == robot.id(); });
+                               [robot](const Robot& r) { return r.id() == robot.id(); });
```

Comment:
What was the reason for this change?

Reply:
The `&` lambda function too much scope. It only needs access to robot, so we can restrict it to that.